h2o-core/src/main/java/water/H2O.java

```diff
@@ -1728,6 +1728,12 @@ public static void gc() {
   // --------------------------------------------------------------------------
   public static void main( String[] args ) {
 
+    String javaVersion = System.getProperty("java.version");
+    if (!(javaVersion.startsWith("1.6") || javaVersion.startsWith("1.7") || javaVersion.startsWith("1.8"))) {
```

Comment:
It is better to check for `null` as well - some obscure JVMs do not need to setup the property.